core/src/main/java/com/onelogin/saml2/util/Util.java

```diff
@@ -1356,14 +1354,13 @@ public static String generateUniqueID() {
 			MessageDigest crypt = MessageDigest.getInstance("SHA-1");
 			crypt.reset();
 			crypt.update(id.getBytes());
-			uniqueIdSha1 = new BigInteger(1, crypt.digest()).toString(16);
+			final String uniqueIdSha1 = new BigInteger(1, crypt.digest()).toString(16);
```

Comment:
Not any reason.

We used a similar approach than [SAMLSSOUtil](https://svn.wso2.org/repos/wso2/carbon/platform/trunk/components/identity/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/util/SAMLSSOUtil.java) 

but
```
uniqueId = "ONELOGIN_" + UUID.randomUUID();;
```
should also works.

The standard says:
```
The mechanism by which a SAML system entity ensures that the identifier is unique is left to the implementation. In the case that a random or pseudorandom technique is employed, the probability of two randomly chosen identifiers being identical MUST be less than or equal to 2^-128 and SHOULD be less than or equal to 2^-160. This requirement MAY be met by encoding a randomly chosen value between 128 and 160 bits in length. The encoding must conform to the rules defining the xs:ID datatype. A pseudorandom generator MUST be seeded with unique material in order to ensure the desired uniqueness properties between different systems
```

Take in mind that UUID internally uses SecureRandom:
[OpenJDK7](http://hg.openjdk.java.net/sandbox/jdk7/jdk/file/3a3c4785e7a7/src/share/classes/java/util/UUID.java#l171) / [OpenJDK6](http://hg.openjdk.java.net/jdk6/jdk6/jdk/file/2d585507a41b/src/share/classes/java/util/UUID.java#l171):